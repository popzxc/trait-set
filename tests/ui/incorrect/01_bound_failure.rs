//! Checks that trait violation is correctly rendered in error.

use trait_set::trait_set;
use std::cell::RefCell;

trait_set! {
    pub trait ThreadSafe = Send + Sync;
}

fn test<T: ThreadSafe>(_t: T) {}

fn main() {
    test(RefCell::new(10u8));
}
