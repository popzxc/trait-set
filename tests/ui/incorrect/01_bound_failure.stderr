error[E0277]: `RefCell<u8>` cannot be shared between threads safely
  --> $DIR/01_bound_failure.rs:13:10
   |
10 | fn test<T: ThreadSafe>(_t: T) {}
   |            ---------- required by this bound in `test`
...
13 |     test(RefCell::new(10u8));
   |          ^^^^^^^^^^^^^^^^^^ `RefCell<u8>` cannot be shared between threads safely
   |
   = help: the trait `Sync` is not implemented for `RefCell<u8>`
   = note: required because of the requirements on the impl of `ThreadSafe` for `RefCell<u8>`
